src/Microsoft.ML.Data/Transforms/ExplainabilityCatalog.cs

```diff
@@ -18,18 +18,25 @@ public static class ExplainabilityCatalog
         /// </summary>
         /// <param name="catalog">The model explainability operations catalog.</param>
         /// <param name="modelParameters">Trained model parameters that support Feature Contribution Calculation and which will be used for scoring.</param>
-        /// <param name="featureColumn">The name of the feature column that will be used as input.</param>
-        /// <param name="numPositiveContributions">The number of positive contributions to report, sorted from highest magnitude to lowest magnitude.
-        /// Note that if there are fewer features with positive contributions than <paramref name="numPositiveContributions"/>, the rest will be returned as zeros.</param>
-        /// <param name="numNegativeContributions">The number of negative contributions to report, sorted from highest magnitude to lowest magnitude.
-        /// Note that if there are fewer features with negative contributions than <paramref name="numNegativeContributions"/>, the rest will be returned as zeros.</param>
+        /// <param name="featureColumnName">The name of the feature column that will be used as input.</param>
+        /// <param name="numberOfPositiveContributions">The number of positive contributions to report, sorted from highest magnitude to lowest magnitude.
+        /// Note that if there are fewer features with positive contributions than <paramref name="numberOfPositiveContributions"/>, the rest will be returned as zeros.</param>
+        /// <param name="numberOfNegativeContributions">The number of negative contributions to report, sorted from highest magnitude to lowest magnitude.
+        /// Note that if there are fewer features with negative contributions than <paramref name="numberOfNegativeContributions"/>, the rest will be returned as zeros.</param>
         /// <param name="normalize">Whether the feature contributions should be normalized to the [-1, 1] interval.</param>
-        public static FeatureContributionCalculatingEstimator FeatureContributionCalculation(this ModelOperationsCatalog.ExplainabilityTransforms catalog,
+        /// <example>
+        /// <format type="text/markdown">
+        /// <![CDATA[
+        /// [!code-csharp[FCT](~/../docs/samples/docs/samples/Microsoft.ML.Samples/Dynamic/FeatureContributionCalculationTransform.cs)]
+        /// ]]>
+        /// </format>
+        /// </example>
+        public static FeatureContributionCalculatingEstimator CalculateFeatureContribution(this TransformsCatalog catalog,
             ICalculateFeatureContribution modelParameters,
-            string featureColumn = DefaultColumnNames.Features,
-            int numPositiveContributions = FeatureContributionDefaults.NumPositiveContributions,
-            int numNegativeContributions = FeatureContributionDefaults.NumNegativeContributions,
+            string featureColumnName = DefaultColumnNames.Features,
```

Comment:
>string featureColumnName = DefaultColumnNames.Features [](start = 12, length = 54)

The `Transformer` that has the model class has both the `ModelParameters` and `FeaturesColumn`. Can we just specify `ITransformer<ICalculateFeatureContribution>` and extract these? As currently phrased, it *is* possible to put the wrong column in and get a runtime error.

Reply:
I double checked with Tom, and he welcomes the change! So I am going to do as you suggest (besides it would be `IPredictionTransformer<ICalculateFeatureCotnribution>`

---
In reply to: [266083168](https://github.com/dotnet/machinelearning/pull/2966#discussion_r266083168) [](ancestors = 266083168)